var/spack/repos/builtin/packages/cbtf-argonavis/package.py

```diff
@@ -67,6 +63,40 @@ class CbtfArgonavis(Package):
 
     parallel = False
 
+    build_directory = 'build_cbtf_argonavis'
+
+    def build_type(self):
+        return 'None'
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Look for package installation information in the cbtf and cbtf-krell
+        # prefixes
+        cmake_prefix_path = join_path(
+            spec['cbtf'].prefix) + ':' + join_path(spec['cbtf-krell'].prefix)
+
+        cmake_args = [
+            '-DCMAKE_PREFIX_PATH=%s' % cmake_prefix_path,
+            '-DCUDA_DIR=%s' % spec['cuda'].prefix,
+            '-DCUDA_INSTALL_PATH=%s' % spec['cuda'].prefix,
+            '-DCUDA_TOOLKIT_ROOT_DIR=%s' % spec['cuda'].prefix,
+            '-DCUPTI_DIR=%s' % spec['cuda'].prefix.extras.CUPTI,
+            '-DCUPTI_ROOT=%s' % spec['cuda'].prefix.extras.CUPTI,
+            '-DPAPI_ROOT=%s' % spec['papi'].prefix,
+            '-DCBTF_DIR=%s' % spec['cbtf'].prefix,
+            '-DCBTF_KRELL_DIR=%s' % spec['cbtf-krell'].prefix,
+            '-DBOOST_ROOT=%s' % spec['boost'].prefix,
+            '-DBoost_DIR=%s' % spec['boost'].prefix,
+            '-DBOOST_LIBRARYDIR=%s' % spec['boost'].prefix.lib,
+            '-DMRNET_DIR=%s' % spec['mrnet'].prefix,
+            '-DBoost_NO_SYSTEM_PATHS=ON']
+
+        # Adjust the standard cmake arguments to what we want the build
+        # type, etc to be
+        self.adjustBuildTypeParams_cmakeOptions(spec, cmake_args)
+        return cmake_args
+
     def adjustBuildTypeParams_cmakeOptions(self, spec, cmakeOptions):
```

Comment:
None of these flags will be in `cmake_args`. `cmake_args` are flags used in addition to the standard flags Spack adds to every package. I would encourage you to let Spack use the standard args and remove this function.